pyccel/codegen/printing/ccode.py

```diff
@@ -625,13 +625,13 @@ def _print_PyccelMod(self, expr):
         second = self._print(expr.args[1])
 
         if expr.dtype is NativeInteger():
-            return "{} % {}".format(first, second)
+            return "((({n} % {base}) + {base}) % {base})".format(n=first, base=second)
 
         if expr.args[0].dtype is NativeInteger():
             first = self._print(NumpyFloat(expr.args[0]))
         if expr.args[1].dtype is NativeInteger():
             second = self._print(NumpyFloat(expr.args[1]))
-        return "fmod({}, {})".format(first, second)
+        return "(fmod(((fmod({n}, {base})) + {base}), {base}))".format(n=first, base=second)
```

Comment:
Could we use the basic expression (e.g. `a % b`) if the second argument is a `Literal`?
e.g.:
```
a % b
a % 2
a % -2
```
would give:
```
(((a % b) + b) % b)
a % 2
(a % 2) + 2
```